engine.py

```diff
@@ -23,200 +19,240 @@
             self.log_error("Nuke 6.3v5 is the minimum version supported!")
             return
-        
-        # keep track of if a UI exists
-        self._ui_enabled = nuke.env.get("gui")
 
-        # versions > 9.0 have not yet been tested so show a message to that effect:
-        if nuke_version[0] > 9 or (nuke_version[0] == 9 and nuke_version[1] > 0):
-            # this is an untested version of Nuke
-            msg = ("The Shotgun Pipeline Toolkit has not yet been fully tested with Nuke %d.%dv%d. "
-                   "You can continue to use the Toolkit but you may experience bugs or "
-                   "instability.  Please report any issues you see to support@shotgunsoftware.com" 
-                   % (nuke_version[0], nuke_version[1], nuke_version[2]))
-            
-            # show nuke message if in UI mode, this is the first time the engine has been started
-            # and the warning dialog isn't overriden by the config:
-            if (self._ui_enabled 
-                and not "TANK_NUKE_ENGINE_INIT_NAME" in os.environ
-                and nuke_version[0] >= self.get_setting("compatibility_dialog_min_version", 10)):
-                nuke.message("Warning - Shotgun Pipeline Toolkit!\n\n%s" % msg)
-                           
-            # and log the warning
-            self.log_warning(msg)
-            
-        # now prepare tank so that it will be picked up by any new processes
+        # Now prepare tank so that it will be picked up by any new processes
         # created by file->new or file->open.
-            
-        # Store data needed for bootstrapping Tank in env vars. Used in startup/menu.py
+        # Store data needed for bootstrapping Tank in env vars. Used in startup/menu.py.
         os.environ["TANK_NUKE_ENGINE_INIT_NAME"] = self.instance_name
         os.environ["TANK_NUKE_ENGINE_INIT_CONTEXT"] = tank.context.serialize(self.context)
         os.environ["TANK_NUKE_ENGINE_INIT_PROJECT_ROOT"] = self.tank.project_path
         
-        # add our startup path to the nuke init path
+        # Add our startup path to the nuke init path
         startup_path = os.path.abspath(os.path.join( os.path.dirname(__file__), "startup"))
         tank.util.append_path_to_env_var("NUKE_PATH", startup_path)        
     
-        # we also need to pass the path to the python folder down to the init script
-        # because nuke python does not have a __file__ attribute for that file
+        # We also need to pass the path to the python folder down to the init script
+        # because nuke python does not have a __file__ attribute for that file.
         local_python_path = os.path.abspath(os.path.join( os.path.dirname(__file__), "python"))
         os.environ["TANK_NUKE_ENGINE_MOD_PATH"] = local_python_path
-        
-    
+
+    @property
+    def has_ui(self):
+        """
+        Whether Nuke is running a GUI session.
+        """
+        return self._ui_enabled
+
+    @property
+    def hiero(self):
```

Comment:
This feels ambiguous to me. Feels like it would be clearer if it said `hiero_enabled` or `hiero_active` or something along those lines to emphasize that it returns a bool.